src/nc/gui/CxxDocument.cpp

```diff
@@ -51,7 +51,7 @@ namespace nc { namespace gui {
 
 CxxDocument::CxxDocument(QObject *parent, std::shared_ptr<const core::Context> context):
     QTextDocument(parent),
-    core::likec::CxxDocument(context),
+    core::likec::CxxDocument(context.get()),
```

Comment:
Could this create an out of scope raw copy of a `std::shared_ptr`? I'm afraid some bad stuff could happen here, too, since I don't understand the way `std::shared_ptr`'s are handed around in this codebase well enough yet.